### PR TITLE
Update authorisation header and extend www-auth with digest

### PR DESCRIPTION
**Pull request by hamishwillee**
This is continued update or HTTP authorisation docs for #8682

This updates the  `WWW-Authenticate` and `Authorization` header docs with additional information. These headers are part of a framework, so the directives depend on the authentication method used. Prior to these updates the headers were written as though only the `Basic` authentication existed.

What I have tried to do is make this framework more obvious by including the directives and examples for both `Basic` and `Digest` authentication. 

Part of the way though though I realised that it is very difficult to do justice to the Digest authentication -  yes I could capture the entire algorithm, but the spec is pretty dense. I don't think we want to go to the point where people think this replaces the spec. What I want to do is give an overview that gives a good flavour of how it works, but makes it pretty clear that you do need to go to the spec if you're going to do anything more than superficial. 

In other words, I don't think this is perfect, but I think it is better. Thoughts?

This part is pretty much done for now. I am planning on updating the BCD https://github.com/mdn/browser-compat-data/pull/12446 to give better links into the spec. That is actually likely to be the most valuable part of this effort :-0

### CODE REVIEW

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 1]
```diff
@@ -103,6 +105,41 @@ For others you may need to check the specifications:
       The only allowed value is the case-insensitive string "UTF-8".
       This does not relate to the encoding of the realm string.
 
+### Digest
+
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/www-authenticate/index.md:109]: This requires careful checking against the spec: https://datatracker.ietf.org/doc/html/rfc7616#page-12

I will be doing that again as I add examples.

File: files/en-us/web/http/headers/authorization/index.md [hunk 2]
```diff
@@ -31,49 +39,52 @@ necessarily, after the server has responded with a {{HTTPStatus("401")}}
...
+
+See also [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) for examples on how to configure Apache or nginx servers to password protect your site with HTTP basic authentication.
 
 ## Specifications
 
 | Specification                                        | Title                                  |
 | ---------------------------------------------------- | -------------------------------------- |
 | {{RFC("7235", "Authorization", "4.2")}} | HTTP/1.1: Authentication               |
 | {{RFC("7617")}}                                 | The 'Basic' HTTP Authentication Scheme |
+| {{RFC("7616")}}                                 | HTTP Digest Access Authentication |
```
Comment (hamishwillee) [inline files/en-us/web/http/headers/authorization/index.md:80]: Note, I have BCD item in progress to add these. Still under discussion. This is kind of "interim solution".

**Review by teoli2003 — APPROVED** [review]

**Comment by hamishwillee** [comment]
It doesn't sit well with me that this you still need to go to the spec to get detail. But it is still better than it was/more useful than over-promoting the basic authentication.

So, merging.
